rts/Lua/LuaOpenGL.cpp

```diff
@@ -5119,6 +5143,50 @@ int LuaOpenGL::ReadPixels(lua_State* L)
 }
 
 
+namespace Impl {
+	template<class Type>
+	static inline void ReadAttachmentPixel(lua_State* L, GLint x, GLint y, GLenum format, GLenum readType, int fSize) {
+		Type data[fSize];
+		glReadPixels(x, y, 1, 1, format, readType, data);
+		for (int e = 0; e < fSize; e++) {
+			lua_pushnumber(L, data[e]);
+		}
+	}
+}
+
+int LuaOpenGL::ReadAttachmentPixel(lua_State* L)
```

Comment:
Why do you need such function? GPU-->CPU read is almost always bad and will cause significant performance implications. If anything consider using PBOs and respective asynchronous API https://riptutorial.com/opengl/example/28872/using-pbos

Reply:
> Why do you need such function? GPU-->CPU read is almost always bad and will cause significant performance implications. If anything consider using PBOs and respective asynchronous API https://riptutorial.com/opengl/example/28872/using-pbos

Do we already have anything like this implemented?

Reply:
> Why do you need such function? GPU-->CPU read is almost always bad and will cause significant performance implications. If anything consider using PBOs and respective asynchronous API https://riptutorial.com/opengl/example/28872/using-pbos

Performance is not priority. Think of this function as a function for a game editor, not the game itself. glReadPixels as it is works incorrectly for integer textures and doesn't allow reading individual attachments from within big FBO.
I don't have experience with PBO. Can you clarify, does it allow to read the result of **this exact frame right now**, or one of the previous ones?

Reply:
> > Do we already have anything like this implemented? Any reference to Recoil code.
> 
> Nope but have a look at this old thread https://community.khronos.org/t/pbo-glreadpixels-not-so-fast/55783/13
> 
> > does it allow to read the result of this exact frame which is happening right now, or it will be one of the previous frames?
> 
> Nope, but usually getting the pixel data a few frames late is fine.

I am aware of glReadPixels performance. If I hit a bottleneck with this, I am going to investigate it further.
The purpose of _this_ function is not fast performant texture read, it is to momentarily read the data from texture, same as gl.ReadPixels, limited to a single pixel, _but_ applied to individual attachment and able to work with integer textures.
This function is a part of an update that enables work with integer textures and individual attachments (glClearBuffer). (Proper setup of integer textures was fixed few PRs ago)

Reply:
Would be good if someone actually tests this later, to find out how slow it really is.